src/framework/ui/tabNavigator/bottomTabNavigator.component.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import {
+  Dimensions,
+  View,
+  ViewProps,
+  StyleSheet,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import {
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+import {
+  TabBarIndicator,
+  Props as TabBarIndicatorProps,
+} from '../tab/tabBarIndicator.component';
+import { Props as ChildProps } from './bottomNavigatorTab.component';
+import { toArray } from '../service/common.service';
+
+type ChildElement = React.ReactElement<ChildProps>;
+
+interface TabNavigatorProps {
+  appearance?: string;
+  selectedIndex?: number;
+  children: ChildElement | ChildElement[];
+  onSelect?: (index: number) => void;
+}
+
+export type Props = TabNavigatorProps & StyledComponentProps & ViewProps;
+
+export class BottomTabNavigator extends React.Component<Props> {
+
+  static defaultProps: Partial<Props> = {
+    selectedIndex: 0,
+  };
+
+  private getComponentStyle = (style: StyleType): StyleType => {
+    return {
+      showText: style.showText,
+      showHighlight: style.showHighlight,
+      container: {
+        backgroundColor: style.backgroundColor,
+        borderTopColor: style.borderTopColor,
+        borderTopWidth: style.borderTopWidth,
+        paddingVertical: style.paddingVertical,
+      },
+      highlight: {
+        height: style.highlightHeight,
+        backgroundColor: style.selectedColor,
+      },
+    };
+  };
+
+  public onSelect = (index: number): void => {
+    if (this.props.selectedIndex !== index) {
+      this.props.onSelect && this.props.onSelect(index);
+    }
+  };
+
+  private getComponentChild(item: ChildElement, index: number): ChildElement {
```

Comment:
Let's unify type. Probably it should be `React.React.Node`?

Reply:
btw ReactNode is a common type which covers even null, undefined and plain text values
I think it's better to use strict types